### ISSUE TITLE
Windows builds are failing

### ISSUE THREAD

**Issue by ocefpaf** [conda-forge/hdf5-feedstock#1]
HDF5 is failing on Windows.

```
BUILD END: hdf5-1.8.16-vc9_0 
TEST START: hdf5-1.8.16-vc9_0 
Fetching package metadata: ........
Solving package specifications: ....
Error: The following specifications were found to be in conflict:
  - hdf5 1.8.16 vc9_0
Use "conda info <package>" to see the dependencies for each package.
Command exited with code 1
```

**Comment by jakirkham** [comment]
That is a very special error. I'm not really sure how we got into this situation.

**Comment by jakirkham** [comment]
I dropped it in the "I need help on Windows" issue so hopefully someone with a dev environment who knows more can figure this out. Were there any recent releases of `conda` as this was working before?

**Comment by jakirkham** [comment]
Oh, maybe we need `python` as a testing requirement. Thoughts? I'll draw up a PR. Feedback welcome. :smile:

**Comment by ocefpaf** [comment]
> Oh, maybe we need `python` as a testing requirement. Thoughts? I'll draw up a PR. Feedback welcome. :smile: 

Makes sense... I guess...

If that works can you update the wiki?

https://github.com/conda-forge/staged-recipes/wiki/VC-features

**Comment by jakirkham** [comment]
Still weird that it worked before though. Maybe there is some subtle difference between CI here and in staged-recipes?

**Comment by msarahan** [comment]
Python should not be required in tests if it is required in run requirements.  However, there is one crazy issue:  Test requirements are NOT pinned the same way that run requirements are: https://github.com/conda/conda-build/issues/835

I can't explain the inconsistent behavior, but it might be related to that issue.

**Comment by jakirkham** [comment]
Slowly coming back to me. We ran into [this issue with GLPK]( https://github.com/conda-forge/staged-recipes/pull/141#issuecomment-198520791 ). Maybe I need to tweak the recipe here to match. That or will need @pelson's CI improvements. Might want to wait to see how GLPK gets fixed before implementing something here. @jjhelmus @pelson, do either of you have any thoughts on this point?

**Comment by jjhelmus** [comment]
At least for GLPK adding `python` as a [run requirement](https://github.com/conda-forge/glpk-feedstock/blob/master/recipe/meta.yaml#L25) on Winodws fixed the build issue.  I agree with the desire for a more elegant solution either with a msvc package or in conda build itself but until then the python run requirement will do the trick.

**Comment by ocefpaf** [comment]
@jakirkham I guess the original issue here is resolved, correct? I saw the logs and there is a new problem:

``` 
C:\conda\conda-bld\test-tmp_dir>where h5c++ 
INFO: Could not find files for the given pattern(s).
 
C:\conda\conda-bld\test-tmp_dir>if errorlevel 1 exit 1 
```

I believe that has a simple solution (missing `.exe` extension).

**Comment by jakirkham** [comment]
So, I think the problem is that executable doesn't exist on Windows. Not sure why that is. However, if you look earlier in the log, you will see that never copied over into the binaries' directory.

**Comment by jakirkham** [comment]
Yeah, appears these are [UNIX only]( https://www.hdfgroup.org/HDF5/Tutor/compile.html ). Will fix the test and then it should be ok.

**Comment by jakirkham** [comment]
Think we have incorporated these fixes. Will see if it builds correctly this time.

**Comment by jakirkham** [comment]
Well, I ran CI under my own user and that fixed almost all issues except for 64-bit support with VS 2008. In other words, we require this fix ( https://github.com/conda-forge/conda-smithy/pull/107 ).

**Comment by jakirkham** [comment]
cc @pelson

**Comment by jakirkham** [comment]
I believe that I have fixed this problem. Here were the problems.

1. VC features were not being followed on Windows during testing.
2. UNIX only binaries were being searched for on Windows.
3. It appears that `h5perf_serial` is missing on Windows.

The first two make sense and have been fixed. Though given the issues with 1, it is possible it will resurface and will keep an eye open for bug reports there to see if we need a better fix. As for 3, this doesn't seem like it should be a problem and it is being built as one can see in the [AppVeyor log]( https://ci.appveyor.com/project/jakirkham/hdf5-feedstock/build/1.0.2/job/j8gxguuawm7mwxvf#L3472 ); however, the [test for it fails]( https://ci.appveyor.com/project/jakirkham/hdf5-feedstock/build/1.0.2/job/j8gxguuawm7mwxvf#L4709 ). Given I don't really have a Windows dev environment, I appreciate any help I can get on that point. My current plan is to inspect the Windows binaries once available to see if this is getting renamed somehow. However, I didn't want to hold up binaries that otherwise seem to work over this one missing command line tool.
